#16 init.sql 추가

0hun opened the pull request:
초기 테이블 정보 추가(회원, 지역, 터미널)

0hun committed:
#16 init.sql 추가

초기 테이블 정보 추가(회원, 지역, 터미널)

0hun committed:
#16 init.sql 추가

account 테이블 인덱스 추가

f-lab-dev commented on src/main/resources/sql/init.sql:
```diff
@@ -0,0 +1,55 @@
+(
+    id           bigint auto_increment comment '아이디'
+        primary key,
+    user_id      varchar(45)  not null comment '회원 아이디',
```
1. 보통 아이디를 이메일로 쓰곤 합니다. 이메일과 통일시켜주는건 어떨까요?
2. 조회를 위해 유니크 인덱스를 걸어주면 좋겠습니다

0hun replied:
넵 알겠습니다.

f-lab-dev commented on src/main/resources/sql/init.sql:
```diff
@@ -0,0 +1,55 @@
+    id           bigint auto_increment comment '아이디'
+        primary key,
+    user_id      varchar(45)  not null comment '회원 아이디',
+    password     varchar(16)  not null comment '패스워드',
```
패스워드가 너무 짧진 않을까요?

0hun replied:
영문특수문자포함해서 16자리까지면 괜찮을꺼같은데 부족할까요?

f-lab-dev replied:
아무래도 패스워드는 넉넉하게 주는게 좋을 것 같습니다~

f-lab-dev commented on src/main/resources/sql/init.sql:
```diff
@@ -0,0 +1,55 @@
+    comment '회원';
+
+create index account_id_password_index
+    on account (id, password);
```
패스워드에 굳이 인덱스를 걸어줘야할까요? 보통 id를 검색하면 패스워드가 같이 올텐데요~

0hun replied:
넵 수정하겠습니다.

f-lab-dev commented on src/main/resources/sql/init.sql:
```diff
@@ -0,0 +1,55 @@
+    on account (id, password);
+
+create index account_id_user_id_name_phone_number_email_index
+    on account (id, user_id, name, phone_number, email);
```
너무 광범위하게 묶어주는 것 같습니다. 이런 검색조건으로는 검색하지 않을 것 같네요~

0hun replied:
넵 수정하겠습니다.

f-lab-dev commented on src/main/resources/sql/init.sql:
```diff
@@ -0,0 +1,55 @@
+    comment '지역';
+
+create index region_id_index
+    on region (id);
```
이미 primary key가 걸려있지 않나요?

0hun replied:
넵 수정하겠습니다.

0hun committed:
#16 리뷰 반영 init.sql 수정

user_id를 email하고 통합하여 email를 id로 사용하도록 크기 늘림
password길이 30자로 수정
user_id 유니크 인덱스 제약조건 추가
account 인덱스 수정
region 인덱스 수정

f-lab-dev commented on src/main/resources/sql/init.sql:
```diff
@@ -2,23 +2,21 @@ create table account
-create index account_id_user_id_name_phone_number_email_index
-    on account (id, user_id, name, phone_number, email);
+create index account_id_user_id_name_index
+    on account (id, user_id, name);
```
userId는 유니크 인덱스를 주었는데 굳이 또 줄 필요가 있나요??

0hun replied:
유니크 인덱스를 걸었을땐, 다시 인덱스를 걸 필요없는거군요

0hun committed:
#16 리뷰 반영 init.sql 수정

account user_id는 이미 유니크 인덱스로 생성되어있기 때문에,
인덱스어서 삭제

0hun committed:
#16 init.sql 추가

user_id를 email로 사용할경우 100자는 짧을수있어 varchar 최대길이로 수정
